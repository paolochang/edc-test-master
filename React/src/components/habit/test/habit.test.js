import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Habit from "../habit";

describe("habit", () => {
  const habit = { id: 1, name: "Reading", count: 1 };

  it("compares snapshot", () => {
    const component = render(
      <Habit
        habit={habit}
        onIncrement={jest.fn()}
        onDecrement={jest.fn()}
        onDelete={jest.fn()}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  describe("calls functions", () => {
    let onIncrement = jest.fn();
    let onDecrement = jest.fn();
    let onDelete = jest.fn();
    let btnIncrement;
    beforeEach(() => {
      render(
        <Habit
          habit={habit}
          onIncrement={onIncrement}
          onDecrement={onDecrement}
          onDelete={onDelete}
        />
      );
    });

    it("handleIncrement", () => {
      btnIncrement = screen.getByTitle("increase");
      userEvent.click(btnIncrement);
      expect(onIncrement).toHaveBeenCalledTimes(1);
    });

    it("handleDecrement", () => {
      btnIncrement = screen.getByTitle("decrease");
      userEvent.click(btnIncrement);
      expect(onIncrement).toHaveBeenCalledTimes(1);
    });

    it("handleDelete", () => {
      btnIncrement = screen.getByTitle("delete");
      userEvent.click(btnIncrement);
      expect(onIncrement).toHaveBeenCalledTimes(1);
    });
  });
});
