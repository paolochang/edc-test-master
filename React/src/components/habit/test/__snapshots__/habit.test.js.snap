// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habit compares snapshot 1`] = `
<div>
  <li
    class="habit"
  >
    <span
      class="habit-name"
      data-testid="habit-name"
    >
      Reading
    </span>
    <span
      class="habit-count"
      data-testid="habit-count"
    >
      1
    </span>
    <button
      class="habit-button habit-increase"
      title="increase"
    >
      <i
        class="fas fa-plus-square"
      />
    </button>
    <button
      class="habit-button habit-decrease"
      title="decrease"
    >
      <i
        class="fas fa-minus-square"
      />
    </button>
    <button
      class="habit-button habit-delete"
      title="delete"
    >
      <i
        class="fas fa-trash"
      />
    </button>
  </li>
</div>
`;
